server/src/main/java/org/uiautomation/ios/utils/SimulatorSettings.java

```diff
@@ -285,47 +286,92 @@ public void resetContentAndSettings() {
         System.err.println("couldn't re-create: " + contentAndSettingsFolder);
       }
     } else {
-      // Starting with Xcode 6 and later, we can use simctl to do the hard work for us.
-      // If it fails, we have to turn the device off first.
       if (!eraseSimulator()) {
-        log.log(Level.WARNING, "Reset content and settings failed, " +
-            "possibly device is in booted state, shutdown device = " + deviceUUID);
-        List<String> simctlArgs = new ArrayList<>();
-        simctlArgs = Arrays.asList("xcrun", "simctl", "shutdown", deviceUUID);
-        Command simctlCmd = new Command(simctlArgs, true);
+        log.info("Erase contents and settings failed on this device: " + deviceUUID);
+        tryToEraseSimulator();
+      }
+    }
+  }
 
-        // Run command 'xcrun simctl shutdown <uuid>'
-        simctlCmd.executeAndWait(false);
+  private void tryToEraseSimulator() {
+    int numTries = 0;
+    boolean successfulReset = false;
+    while ((!successfulReset) && (numTries < NUMBER_TRIES_RESET_DEVICE)) {
+      log.info("Wait " + SLEEP_TIME_BETWEEN_RESET_TRIES + " milliseconds before reattempt.");
+      try {
+        Thread.sleep(SLEEP_TIME_BETWEEN_RESET_TRIES);
+      } catch (InterruptedException e) {
+        // ignored
+      }
+      successfulReset = eraseSimulator();
+      numTries += 1;
+    }
+    if (!successfulReset) {
+      int totalWaitTime = SLEEP_TIME_BETWEEN_RESET_TRIES * NUMBER_TRIES_RESET_DEVICE;
+      String message = "Erase contents and settings still failed. Total waiting time: "
+              + totalWaitTime
+              + ". Now try to shutdown device: " + deviceUUID;
+      log.warning(message);
 
-        // Retry
-        eraseSimulator();
+      Boolean successfulShutdown = shutdownDevice();
+      if (successfulShutdown) {
+        successfulReset = eraseSimulator();
       }
+
+      if (!successfulReset) {
+        message = "Unable to erase contents and settings of this device: " + deviceUUID;
+        log.warning(message);
+        // add more information to exception message
+        message += ". Tried " + NUMBER_TRIES_RESET_DEVICE + " times with with a waiting period of " + SLEEP_TIME_BETWEEN_RESET_TRIES
+                + " millisecond between each attempt. ";
+        if (!successfulShutdown) {
+          message += "Also unable to shut down the device.";
+        }
+        throw new WebDriverException(message);
+      }
+    }
+  }
+
+  private boolean shutdownDevice() {
+    List<String> simctlArgs;
+    simctlArgs = Arrays.asList("xcrun", "simctl", "shutdown", deviceUUID);
+    Command simctlCmd = new Command(simctlArgs, true);
+    int exitCode = simctlCmd.executeAndWait(true);
+
+    // check for success codes
+    if (exitCode == 146) {
```

Comment:
does anyone know what `146` means?   I'm guessing that it may refer to *128 + signum* which would imply signal 18 (`SIGTSTP`) which doesn't make much sense to me.